packages/react-divider/Spec.md

```diff
@@ -0,0 +1,222 @@
+The purpose of redoing the component is to add minor structural updates and new engineering practices and update the naming conventions in the API to match naming conventions indicated by OpenUI research.
+
+## Prior Art
+
+[Github Epic Convergence Issue #16254](https://github.com/microsoft/fluentui/issues/16254)
+
+[Github Convergence Issue #15759](https://github.com/microsoft/fluentui/issues/15759)
+
+[Fluent UI Separator](https://developer.microsoft.com/en-us/fluentui#/controls/web/separator)
+
+[Open UI Component Matrix](https://open-ui.org/analysis/component-matrix)
+
+[Ant Design Divider](https://ant.design/components/divider/)
+
+[Fast Divider](https://explore.fast.design/components/fast-divider)
+
+[Semantic UI Divider](https://semantic-ui.com/elements/divider.html)
+
+## Comparison of v8 and v0
+
+- v8 - [Separator](https://developer.microsoft.com/en-us/fluentui#/controls/web/separator)
+- v0 - [Divider](https://fluentsite.z22.web.core.windows.net/0.51.4/components/divider/definition)
+
+## Sample Code
+
+Basic Examples:
+
+```html
+<Divider />
+<Divider vertical />
+<Divider>This is a divider with content</Divider>
+```
+
+## Variants
+
+The divider will have two main variants, rendering horizontally or vertically. The control is the same with the optional property `vertical` defining how it renders.
+
+Standard default horizontal
+
+```html
+<Divider />
+```
+
+Vertical
+
+```html
+<Divider vertical />
+```
+
+Color
+
+```html
+<Divider color="black" />
+```
+
+Content
+
+```html
+<Divider>My Content</Divider>
+```
+
+Icon
+
+```html
+<Divider><Icon /></Divider>
+```
+
+Important
+
+```html
+<Divider important>This is important!</Divider>
+```
+
+Inset
+
+```html
+<Divider inset />
+```
+
+Appearance
+
+```html
+<Divider apperance="subtle" />
+<Divider apperance="brand" />
+<Divider apperance="strong" />
+```
+
+## API
+
+From [Divider.types.tsx](http://about:blank)
+
+### Slots
+
+- content (?)
+
+### Props
+
+```ts
+export interface DividerProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  tokens?: RecursivePartial<DividerTokens>;
+
+  /**
+   * Determines the alignment of the content within the divider.
+   * @defaultvalue 'center'
+   */
+  alignContent?: 'start' | 'end' | 'center';
+
+  appearance?: 'default' | 'subtle' | 'brand' | 'strong';
+
+  children?: any;
+
+  /* A divider can have a overriding border color */
+  color?: string;
+
+  /* A divider can be classified as important to emphasize its content */
+  important?: boolean;
+
+  inset?: boolean;
+
+  /* A divider can be horizontal (default) or vertical*/
+  vertical?: boolean;
+
+  /* Overrides for custom appearances */
+  height?: string;
+  width?: string;
+
+  fontSize?: string;
+  fontWeight?: string;
+  fontColor?: string;
+
+  margin?: string;
+  marginTop?: string;
+  marginBottom?: string;
+  marginLeft?: string;
+  marginRight?: string;
+
+  borderStyle?: string;
+  borderSize?: string | number;
```

Comment:
I don't think that it's expected that we will include CSS properties to props interface. Some of them can be tweaked with component tokens, but this story is still in progress...

Is it intentional to have these props?

Reply:
This was tested and without them being defined, it creates a compile error as they are used / referenced in the useDividerStyles.tsx. I spoke with Mak and until there is a means of working around this, he mentioned to leave them.

Reply:
I was telling @joschemd exactly that @layershifter, however, some of his customizations are based on them and we don't have a full story for component tokens. I'd say we can leave them now but possibly add a big comment noting that they will be removed once the component tokens story is done.

I also think that while leaving them in `Divider.types.ts` should be fine with such a comment we should remove them from the spec since it is just an interim step and not where we're intending to ultimately go.

Reply:
Updated